packages/babel-plugin-transform-es2015-modules-commonjs/src/index.js

```diff
@@ -350,7 +358,15 @@ export default function () {
           for (let source in imports) {
             let {specifiers, maxBlockHoist} = imports[source];
             if (specifiers.length) {
-              let uid = addRequire(source, maxBlockHoist);
+              let uid;
+
+              if (hasSingleDefaultImport(specifiers)) {
+                uid = addRequire(source, maxBlockHoist, {
+                  inlineDefault: this.addHelper("interopRequireDefault")
+                });
+              } else {
+                uid = addRequire(source, maxBlockHoist);
+              }
```

Comment:
Could you validate this with a file like

```
import Foo from 'foo';
import {named} from 'foo';

console.log(Foo, named);
```

I think at the moment we merge the `require` calls for multiple imports with the same source, so this might be an issue.